clang/lib/Driver/Driver.cpp

```diff
@@ -767,6 +768,27 @@ Driver::OpenMPRuntimeKind Driver::getOpenMPRuntime(const ArgList &Args) const {
   return RT;
 }
 
+static const char *getDefaultSYCLArch(Compilation &C) {
+  // If -fsycl is supplied we will assume SPIR-V
+  if (C.getDefaultToolChain().getTriple().getArch() == llvm::Triple::x86)
```

Comment:
If I get it right, here we try to check the address size on the host to propagate the same size for the device compilation.
If so, I suggest we generalize it for more host targets:

```suggestion
  if (C.getDefaultToolChain().getTriple().isArch32Bit())
```

Reply:
Yup, that's right.  I've updated to use your suggestion.